Issue #282 New Notch options

DSCaskey opened the pull request:
Fix issue # 282 Create new Notch Options
Tested on Windows 8 64 bit
Tested on Windows 10 64-bit

DSCaskey committed:
Merge branch 'issue-#183' into develop

DSCaskey committed:
Fix issue 141: Crash when using Window->Close pattern menu

This fixes the Bad Id exception caused  when trying to close a pattern using the Window->Close pattern menu item.

DSCaskey committed:
Issue #282 / Create new Notch options..

This adds new notch options and completes the change of the term "passmarks" to "notch", and the change of the text "typeLine" to "lineType" within the schema  conversion from ver 0.6.0 to ver 0.6.1.

slspencer commented:
Hey @DSCaskey 
Did you see this error in the build
> make[3]: *** No rule to make target 'share/resources/icon/24x24/u_notch.png', needed by 'qrc_icon.cpp'.  Stop.

Have you seen this article [How to embed an icon in a Qt application](https://www.walletfox.com/course/qtresourceexample.php)?

slspencer reviewed (changes requested):
Build error   
> make[3]: *** No rule to make target 'share/resources/icon/24x24/u_notch.png', needed by 'qrc_icon.cpp'. Stop.

The icon isn't referenced in a Qt resource file. This article discusses the baroque steps needed to accomplish this task: 

>https://www.walletfox.com/course/qtresourceexample.php

DSCaskey commented:
> Did you see this error in the build

Hey Susan,

Yes I saw it in my email on my phone next day @ work, but didn't have my github password on the phone to see the details until now.

Yes I know how to add files to the resources. BTW. - that link is outdated... it;s for Designer which has been supplanted with Creator. You can simply add files to a qrc using the context menu without having to edit the qrc. 

![qrc](https://user-images.githubusercontent.com/31944718/90951673-cbf7de80-e42a-11ea-96c5-96ccf823be20.png)



Just looked at it now. I have no clue - as it's compiling fine on both my laptops 5.14.2 & 5.15 (which I just installed).

 The u_notch.png file is there in the folder, in the qrc, and in the running program? 

![unotch](https://user-images.githubusercontent.com/31944718/90951446-0e202080-e429-11ea-9a22-f9bf25340298.png)


What I do know is that usually when you get a "No rule to make target " it means you have to clean a [sub]project when you've added files after you compiled a .pro...and the make rules are not updated to reflect the added file. An after cleaning you  sometimes have to close & reopen the project so the qrc is re read. 

I assume though that Github is doing a clean build?.... so I I'm kinda of at a loss.... at the moment.

DSCaskey commented:
Hey Susan,

I think I figured out what happened. The new notch icons are not committed to repo, even though they exist in the qrc file. I just dragged a copy into the 24x24 folder - which by itself does not commit them, and Creator did't  commit them when I added them to the project file. Doh. 

I'll commit the png additions and push the changes.

DSCaskey committed:
Add new notch icons

DSCaskey committed:
Fix test for VPiece

Fix test for VPiece to allow use of VCommonSettings in a piece node. Also fix a few anomalies in the drawing of notches.

DSCaskey committed:
Fix slash in include file

Replace forward slash with backward slash in include file name

DSCaskey committed:
Delete .qmake.stash

DSCaskey committed:
Delete object_script.seamly2d

DSCaskey committed:
Delete object_script.seamlyme

DSCaskey committed:
Delete object_script.libifc

DSCaskey committed:
Delete object_script.qmuparser

DSCaskey committed:
Delete qmuparser_resource.rc

DSCaskey committed:
Delete object_script.libvdxf

DSCaskey committed:
Delete object_script.libvgeometry

DSCaskey committed:
Delete object_script.libvlayout

DSCaskey committed:
Delete object_script.libvmisc

DSCaskey committed:
Delete object_script.libvpatterndb

DSCaskey committed:
Delete object_script.libvwidgets

DSCaskey committed:
Delete object_script.vpropertyexplorer.Debug

DSCaskey committed:
Delete object_script.vpropertyexplorer.Release

DSCaskey committed:
Delete object_script.libvtools
